src/pages/learn/introspection.mdx

```diff
@@ -146,25 +144,36 @@ tell us what these are lists of.
 }
 ```
 
-Let's end with a feature of the introspection system particularly useful
-for tooling; let's ask the system for documentation!
+Let's end with a feature of the introspection system particularly useful for tooling; let's ask the system for documentation:
 
 ```graphql
 # { "graphiql": true }
-{
+query {
   __type(name: "Droid") {
     name
     description
   }
 }
 ```
 
-So we can access the documentation about the type system using introspection,
-and create documentation browsers, or rich IDE experiences.
+As demonstrated above, we can access the documentation about the type system using introspection and create documentation browsers or rich IDE experiences.
+
+This has just scratched the surface of the introspection system; we can query for Enum type values, what Interface types another type implements, and more. We can even introspect on the introspection system itself.
+
+To see an example of a specification-compliant GraphQL query introspection system implemented in code, you can view the [introspection.ts](https://github.com/graphql/graphql-js/blob/main/src/type/introspection.ts) file in GraphQL.js repository.
+
+## Introspection in production
+
+Introspection is a useful feature of GraphQL, especially for client developers. However, for private GraphQL APIs, it may be undesirable to expose sensitive information about the schema publicly via the `__schema` field in a production environment.
+
+In those cases, it may be preferable to disable introspection in production environments as a part of a broader API security strategy that could also include authentication and authorization, depth-limiting, query cost analysis, and more.
```

Comment:
We should not promote disabling introspection as a way to prevent exposing sensitive information about the schema. The schema should not _have_ sensitive information in it. However, it is reasonable to disable introspection to reduce the attack surface. I've rewritten these two paragraphs to clarify what is meant by a "private GraphQL API" (by avoiding use of that term) and have rewritten to encourage disabling introspection in production without suggesting that doing so hides sensitive information. I've also highlighted that you can either do safe-listing or you can do a number of other techniques and hope for the best.

```suggestion
Introspection is a useful feature of GraphQL, especially for client developers and tooling. However, for APIs intended only for your own applications, it’s typically not needed in production—required operations are usually baked into these applications at build time, making runtime introspection unnecessary.

Disabling introspection in production is common in order to reduce the API’s attack surface. This is often part of a broader API security strategy, which may also include authentication and authorization, operation safe-listing (or a range of alternative protections, such as depth-limiting, breadth-limiting, alias limits, cycle rejection, cost analysis, etc.), execution timeouts, and more.
```